strings/base_events.h

```diff
@@ -342,14 +342,9 @@ namespace winrt::impl
         return 1;
     }
 
-    template <typename Delegate, typename... Arg>
-    bool invoke(Delegate const& delegate, Arg const&... args) noexcept
+    struct report_helper
     {
-        try
-        {
-            delegate(args...);
-        }
-        catch (...)
+        WINRT_IMPL_NOINLINE static bool report_failed_invoke()
```

Comment:
This seems overly complicated. The cppwinrt approach is to use `inline WINRT_IMPL_NOINLINE` on a free function the same way `winrt::throw_hresult` is written.

Reply:
I was innundated with compiler errors about "removing unreferenced method." The way around that was `WINRT_IMPL_NOINLINE inline ...` which seemed weird. I'll try again.